### PR TITLE
Add release notes for .NET provider 7.2.0 release

### PR DESCRIPTION
**Pull request by amerjusupovic**

### CODE REVIEW

File: releaseNotes/MicrosoftExtensionsConfigurationAzureAppConfiguration.md [hunk 1]
```diff
@@ -1,6 +1,14 @@
 # Microsoft.Extensions.Configuration.AzureAppConfiguration
 [Source code][source_code] | [Package (NuGet)][package]
 
+## 7.2.0 - May 2, 2024
+### Enhancements
+* Added support for .NET 8 as a target framework. [#530](https://github.com/Azure/AppConfiguration-DotnetProvider/issues/530)
+* Removed all usage of reflection for JSON parsing, allowing native AOT applications to use JSON key-values from Azure App Configuration with the .NET provider. [#537](https://github.com/Azure/AppConfiguration-DotnetProvider/issues/537)
+
+### Bug Fixes
+* Updated `Azure.Data.AppConfiguration` reference to `1.4.1` to fix a bug where responses from Azure App Configuration with status code 304 would show up as failures in Application Insights. [#514](https://github.com/Azure/AppConfiguration-DotnetProvider/issues/514).
```
Comment (avanigupta) [inline releaseNotes/MicrosoftExtensionsConfigurationAzureAppConfiguration.md:10]: Its not really this library's bug. You could just say that the SDK version was updated.

File: releaseNotes/MicrosoftAzureAppConfigurationFunctionsWorker.md [hunk 2]
```diff
@@ -1,6 +1,11 @@
 # Microsoft.Azure.AppConfiguration.Functions.Worker
 [Source code][source_code] | [Package (NuGet)][package]
 
+## 7.2.0 - May 2, 2024
+### Enhancements
+* Added support for .NET 8 as a target framework. [#530](https://github.com/Azure/AppConfiguration-DotnetProvider/issues/530)
```
Comment (avanigupta) [inline releaseNotes/MicrosoftAzureAppConfigurationFunctionsWorker.md:6]: We normally just mention this in the base package release notes.

File: releaseNotes/MicrosoftExtensionsConfigurationAzureAppConfiguration.md [hunk 3]
```diff
@@ -1,6 +1,14 @@
 # Microsoft.Extensions.Configuration.AzureAppConfiguration
 [Source code][source_code] | [Package (NuGet)][package]
 
+## 7.2.0 - May 2, 2024
+### Enhancements
+* Added support for .NET 8 as a target framework. [#530](https://github.com/Azure/AppConfiguration-DotnetProvider/issues/530)
+* Removed all usage of reflection for JSON parsing, allowing native AOT applications to use JSON key-values from Azure App Configuration with the .NET provider. [#537](https://github.com/Azure/AppConfiguration-DotnetProvider/issues/537)
```
Comment (avanigupta) [inline releaseNotes/MicrosoftExtensionsConfigurationAzureAppConfiguration.md:7]: How about this:
"Removed reflection-based JSON serialization to enable native AOT applications to utilize JSON key-values from Azure App Configuration."

File: releaseNotes/MicrosoftExtensionsConfigurationAzureAppConfiguration.md [hunk 4]
```diff
@@ -1,6 +1,14 @@
...
+
+### Bug Fixes
```
Comment (avanigupta) [inline releaseNotes/MicrosoftExtensionsConfigurationAzureAppConfiguration.md:9]: Not a bug fix, it can go in the enhancements section

**Review by avanigupta — APPROVED** [review]

File: releaseNotes/MicrosoftAzureAppConfigurationFunctionsWorker.md (hunk 2 above)
Comment (zhenlan) [reply releaseNotes/MicrosoftAzureAppConfigurationFunctionsWorker.md]: Correct for the base, but it is a change for this package too.

Comment (avanigupta) [reply releaseNotes/MicrosoftAzureAppConfigurationFunctionsWorker.md]: That's true. We've just been following this pattern in our previous release notes. However, I'm also fine with adding it to all release notes going forward.

Comment (zhenlan) [reply releaseNotes/MicrosoftAzureAppConfigurationFunctionsWorker.md]: Yes, we can do it more explicitly going forward.

**Review by zhenlan — APPROVED** [review]

**Pull request merged**